routers/web/repo/view.go

```diff
@@ -495,6 +495,11 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		ctx.Data["ReadmeExist"] = readmeExist
 
 		markupType := markup.Type(blob.Name())
+		detected := false
+		if markupType == "" {
+			detected = true
+			markupType = markup.DetectRendererType(bytes.NewReader(buf))
```

Comment:
You don't know how large is this file. A text maybe very large.

Reply:
it's known, it will be at most 1024 bytes:
https://github.com/go-gitea/gitea/pull/19497/files#diff-43a7f4747c7ba8bff888c9be11affaafd595fd55d27f3333840eb19df9fad393R262